src/main/java/target/HttpTarget.java

```diff
@@ -81,11 +80,38 @@ public CompletableFuture<TargetResponse> call(final ConsumerRecord<String, Strin
                         }
                     }
                 );
+
+            var result = new CompletableFuture<Optional<HttpResponse<String>>>();
+
+            httpFuture.whenComplete(
+                (ok, error) -> {
+                    if (error != null) {
+                        result.completeExceptionally(error);
+                    } else {
+                        result.complete(Optional.of(ok));
+                    }
+                }
+            );
+
+            return result;
+        };
+
+        Optional<HttpResponse<String>> defaultResponse = Optional.empty();
+
         return Failsafe
-            .with(retryPolicy.<HttpResponse<String>>get(record, HttpResponse::statusCode))
+            .with(
+                Fallback.of(defaultResponse),
+                retryPolicy.get(record, o -> o.map(r -> r.statusCode()).orElseGet(() -> -1))
+            )
             .getStageAsync(completionStageCheckedSupplier)
             .thenApplyAsync(
-                response -> {
+                optionalResponse -> {
+                    if (optionalResponse.isEmpty()) {
```

Comment:
If I understand correctly this is the only change needed for the problem you are trying to solve.

I really prefer to do as little as possible in a PR of this infra library. There are other users to it and small PRs will reduce risk significantly l